src/extension.ts

```diff
@@ -567,6 +567,43 @@ function launchMetals(
         );
       });
 
+      function displayBuildTarget(target: string): void {
+        const workspaceRoots = workspace.workspaceFolders;
+        if (workspaceRoots && workspaceRoots.length > 0) {
+          const uriStr = `metalsDecode:file:///${workspaceRoots[0].name}/${target}.metals-buildtarget`;
+          const uri = Uri.parse(uriStr);
+          workspace
+            .openTextDocument(uri)
+            .then((textDocument) => window.showTextDocument(textDocument));
+        }
+      }
+
+      registerCommand(`metals.target-info-display`, async (...args: any[]) => {
+        if (args.length > 0) {
+          // get build target name from treeview uri of the form "projects:file:/root/metals/.mtags/?id=mtags3!/_root_/,"
+          const treeViewUri = args[0] as string;
+          const buildTargetRegEx = new RegExp("projects:.*?id=(.*)!.*", "g");
```

Comment:
Could we instead parse it as an URI and get the query parameters?

Reply:
It would still only get us halfway there.

A URI example from the treeview would be...
`projects:file:/root/metals/.mtags/?id=mtags3!/_root_/`

Using `URLSearchParams(uri).get("id")` would give...
`mtags3!/_root_/`

So I'd still need to parse for `!` to get `mtags3`

I can do that if you want?

Reply:
We could then just do `split("!")` so maybe it's a bit clrearer than a regex?

Reply:
This splitting part seems too much for me.
IMO in this case regex is simpler and with the proper example in the comment (exists) isn't also that hard to understand.

Reply:
I'm easy.  Ideally the treeviewuri would be in the format we already want but it's used to work out how to drill down into classes/methods so I wasn't going to change it.  I tried to extract the target name with `URLSearchParams` but it doesn't like our URI so converts the whole thing to a single query param which is no help. I guess URI != URL

Reply:
ok then, for me whichever is fine

Reply:
I've changed to `split` + `split`.  Does the job. 🤷